### PR TITLE
[SYCLomatic] Refine migration of 19 thrust APIs with Rewriter migration framework

### PR DESCRIPTION
**Pull request by tomflinda**
Rewrite thrus APIs list are as follows:
thrust::max_element
thrust::min_element
thrust::unique
hrust::count
thrust::generate
thrust::generate_n
thrust::merge
thrust::upper_bound 	
thrust::lower_bound 	
thrust::binary_search
thrust::fill
thrust::fill_n
thrust::count_if
thrust::raw_pointer_cast
thrust::sort
thrust::sequence 
thrust::make_discard_iterator
thrust::make_constant_iterator
thrust::make_counting_iterator

 Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

**Review by wenhuiNi — APPROVED** [review]

File: clang/lib/DPCT/CallExprRewriterThrust.cpp [hunk 1]
```diff
@@ -127,7 +165,7 @@ makeMappedThrustPolicyEnum(unsigned Idx) {
                  getDrefName(CMCE->getArg(0)) + ")";
         }
       }
-    } else if (auto CE = dyn_cast<CallExpr>(E)) {
+    } else if (auto CE = dyn_cast<CallExpr>(C)) {
```
Comment (zhiweij1) [inline clang/lib/DPCT/CallExprRewriterThrust.cpp]: why change `E` to `C`?
`C` is `CallExpr`, the `dyn_cast` is unnecessary.
`E` is the argument expression.

**Review by zhiweij1 — CHANGES_REQUESTED** [review]

File: clang/lib/DPCT/CallExprRewriterThrust.cpp (hunk 1 above)
Comment (tomflinda) [reply clang/lib/DPCT/CallExprRewriterThrust.cpp]: Good catch, should be E instead. 
while this branch is dead code now, will be removed in the updated PR.

**Review by zhiweij1 — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Review by AndyCHHuang — APPROVED** [review]
LGTM

**Pull request merged**